### ISSUE TITLE
HTML seems to be exported without paragraphs or new lines for single line break

### ISSUE THREAD

**Issue by jmartinesp** [matrix-org/matrix-rich-text-editor#862]
I was doing some testing on Android when I got this result:

![image](https://github.com/matrix-org/matrix-rich-text-editor/assets/480955/30c009b5-a32a-4453-932c-bf5219c2c1a6)

Above is the composer, below is a text view with the result from the HTML generated above. I plugged in the debugger and I found the composer rendered this:

```html
<ol><li>test</li></ol><p>hmmm</p>
```

While the html passed to the text view for rendering was:
```html
<ol><li>test</li></ol>hmmm
```

This happens after all block elements (lists, code blocks, quotes, etc.).

**Comment by jonnyandrew** [comment]
The difference in the HTML between the editor and the text view is expected; what you see in the editor is the internal representation whereas the text view is being fed the stripped down HTML which is formatted as if it is a message. Hence why the text view may contain `<br />` and the editor contains empty `<p>`s.

So the issue here (I think) is that the text view is not rendering a line break after the `<ol>` block?

**Comment by jmartinesp** [comment]
In the example above, the HTML fed to the EditText has paragraphs, but the HTML fed to the TextView has nothing, no line breaks nor paragraphs. It's not the TextView that's not rendering it, it's that it never received it in the first place.

**Comment by jmartinesp** [comment]
> So the issue here (I think) is that the text view is not rendering a line break after the <ol> block?

Yeah sorry, I didn't realise the RTE library was translating something like `<p>Test</p>` to `Test<br />` and not `<br />Test`.  There's probably some missing case for handling the end of a list in the Rust code.

**Comment by jonnyandrew** [comment]
Are you sure this is an issue with the HTML? My understanding was that there should be an implicit line break after a block in HTML: https://jsfiddle.net/78nujox1/

**Comment by jmartinesp** [comment]
> Are you sure this is an issue with the HTML? My understanding was that there should be an implicit line break after a block in HTML: [jsfiddle.net/78nujox1](https://jsfiddle.net/78nujox1/)

I think you're right, thanks for the example. In my mind we could only mix block elements with block elements, but in HTML you can mix block elements and inline ones, creating 'virtual' block ones. I guess we'll need to adapt block elements like this to render an extra line break in Android.
